improver/field_texture.py

```diff
@@ -0,0 +1,251 @@
+                field's texture. A larger radius should be used for diagnosing larger-scale
+                textural features.
+
+            textural_threshold (float):
+                A unit-less threshold value that defines the ratio value above which
+                the field is considered rough and below which the field is considered
+                smoother.
+
+            diagnostic_threshold (float):
+                A user defined threshold value related either to cloud or precipitation,
+                used to extract the corresponding dimensional cube (default: 1).
+
+        """
+        self.nbhood_radius = nbhood_radius
+        self.textural_threshold = textural_threshold
+        self.diagnostic_threshold = diagnostic_threshold
+        self.cube_name = None
+
+    def _calculate_ratio(self, cube, radius):
+        """
+        Calculates the ratio of actual to potential value transitions in a
+        neighbourhood about each cell.
+
+        The process is as follows:
+
+            1. For each grid cell find the number of cells of value 1 in a surrounding
+               neighbourhood of a size defined by the arg radius. The potential
+               transitions within that neighbourhood are defined as the number of
+               orthogonal neighbours (up, down, left, right) about cells of value 1.
+               This is 4 times the number of cells of value 1.
+            2. Calculate the number of actual transitions within the neighbourhood,
+               that is the number of cells of value 0 that orthogonally abut cells
+               of value 1.
+            3. Calculate the ratio of actual to potential transitions.
+
+        Ratios approaching 1 indicate that there are many transitions, so the field
+        is highly textured (rough). Ratios close to 0 indicate a smoother field.
+
+        A neighbourhood full of cells of value 1 will return ratios of 0; the
+        diagnostic that has been thresholded to produce the binary field is found
+        everywhere within that neighbourhood, giving a smooth field. At the other
+        extreme, in neighbourhoods in which there are no cells of value 1 the ratio
+        is set to 1.
+
+        Args:
+            cube (iris.cube.Cube):
+                Input data in cube format containing a two-dimensional field
+                of binary data.
+
+            radius (float):
+                Radius for neighbourhood in metres.
+
+        Returns:
+            iris.cube.Cube:
+                A ratio between 0 and 1 of actual transitions over potential transitions.
+        """
+        # Calculate the potential transitions within neighbourhoods.
+        potential_transitions = SquareNeighbourhood(sum_or_fraction="sum").run(
+            cube, radius=radius
+        )
+        potential_transitions.data = np.where(
+            cube.data > 0, potential_transitions.data * 4, 0
+        )
```

Comment:
As discussed, please could you change this to:
```
potential_transitions.data = 4 * potential_transitions.data
```

Reply:
Done.